osu.Game/Beatmaps/Drawables/UpdateableBeatmapSetCover.cs

```diff
@@ -9,79 +9,45 @@
 
 namespace osu.Game.Beatmaps.Drawables
 {
-    public class UpdateableBeatmapSetCover : Container
+    public class UpdateableBeatmapSetCover : ModelBackedDrawable<BeatmapSetInfo>
     {
-        private Drawable displayedCover;
-
-        private BeatmapSetInfo beatmapSet;
+        private readonly BeatmapSetCoverType coverType;
 
         public BeatmapSetInfo BeatmapSet
         {
-            get => beatmapSet;
-            set
-            {
-                if (value == beatmapSet) return;
-
-                beatmapSet = value;
-
-                if (IsLoaded)
-                    updateCover();
-            }
+            get => Model;
+            set => Model = value;
         }
 
-        private BeatmapSetCoverType coverType = BeatmapSetCoverType.Cover;
-
-        public BeatmapSetCoverType CoverType
+        public new bool Masking
```

Comment:
Why are `new`-ing this?  I don't see the need for this redeclaration when `base.Masking` is public, and you're not doing anything more than calling `base`.

Reply:
this is resharper enforced, for clarity when working with the class itself (sometimes you intentionally want to call `base.`).

Reply:
My point was more why does the property exist at all.

Reply:
`ModelBackedDrawable` is a composite, which doesn't expose masking itself. That said, you may have a point – could we not just set `Masking` via protected in all use cases? Seems saner to me.

Reply:
`CompositeDrawable.Masking` is public.

Reply:
actually, i think it would be better if i did the masking inside `BeatmapSetCover`